File: 0349 intersection-of-two-arrays/solution.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<bool> s(1001,false);
        for(int i:nums1) s[i]=true;
        vector<int> ans;
        for(int i:nums2) {
            if(s[i]) {
                ans.push_back(i);
                s[i]=false;
            }
        }
        return ans;
    }
};

File: 0349 intersection-of-two-arrays/solution.cs
public class Solution {
    public int[] Intersection(int[] nums1, int[] nums2) {
        bool[] b = new bool[1001];
        foreach(int i in nums1) b[i]=true;
        int[] ans = new int[nums1.Length];
        int idx =0;
        foreach(int i in nums2) {
            if(b[i]) {
                ans[idx++] = i;
                b[i]=false;
            }
        }
        int[] result = new int[idx];
        Array.Copy(ans, result, idx); 
        return result;
    }
}
